src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java

```diff
@@ -87,6 +115,96 @@ public String toString(){
                 }
             }
         }
+        if (!compressedFiles.isEmpty()) {
+            try {
+                fileList.addAll(extractCompressedFiles(compressedFiles));
+            } catch (Exception e) {
+                e.printStackTrace();
+                if (LOG.isErrorEnabled()) {
+                    LOG.error("Skipping files to extract.");
+                }
+            }
+        }
         return fileList;
     }
+
+    public static List<File> extractCompressedFiles(List<File> fileList) throws Exception {
+        temp_dir = createTempDirectory();
+        List<File> extracted = new ArrayList<>();
+        if (!fileList.isEmpty()) {
+            for (File extractFile : fileList) {
+                if (isCompressedFile(extractFile)) {
+                    File tmp_file;
+                    try {
+                        tmp_file = File.createTempFile(extractFile.getName(), "", temp_dir);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                        if (LOG.isErrorEnabled()) {
+                            LOG.error("Error creating temp file in : " + temp_dir.getAbsolutePath() + "\nSkipping File");
+                        }
+                        continue;
+                    }
+                    try {
+                        BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(tmp_file));
+                        FileInputStream fis = new FileInputStream(extractFile);
+                        if (extractFile.getName().endsWith(COMPRESSION_FORMAT.GZIP.toString())) {
+                            GZIPInputStream gzip = new GZIPInputStream(new BufferedInputStream(fis));
+                            int readByte;
+                            while ((readByte = gzip.read()) > 0) {
+                                bos.write(readByte);
+                            }
+                            gzip.close();
+                            Path path = Files.move(Paths.get(tmp_file.getAbsolutePath()), Paths.get(temp_dir.getAbsolutePath(), extractFile.getName().replace(COMPRESSION_FORMAT.GZIP.toString(), "")));
+                            extracted.add(new File(path.toUri()));
+                        }
+                        bos.close();
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        deleteTempDir();
+                        throw new Exception("Error while decompressing files");
+                    }
+                }
+            }
+        }
+        return extracted;
+    }
+
+    private static boolean isCompressedFile(File extractFile) {
+        return extractFile.getName().endsWith(COMPRESSION_FORMAT.GZIP.toString());
+    }
+
+    public static void deleteTempDir() {
```

Comment:
I would use an open source library to perform IO operations like this.  I would suggest [Apache Commons FileUtils](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/index.html?org/apache/commons/io/FileUtils.html)